Destructuring assignment cause JVM crash when declared inside a closure

```golo
#!/usr/bin/env golosh
module test

function test = {
  return {
    let a, b = [1,2] # paf!
    return a
  }
}

function main = |args| {
  let x = test()()
  println(x)
}
```
The code above seem's not generate correct bytecode:

```
Exception in thread "main" java.lang.IllegalAccessException: no such method: test.main(String[])void/invokeStatic
        at java.lang.invoke.MemberName.makeAccessException(MemberName.java:869)
        at java.lang.invoke.MemberName$Factory.resolveOrFail(MemberName.java:1005)
        at java.lang.invoke.MethodHandles$Lookup.resolveOrFail(MethodHandles.java:1382)
        at java.lang.invoke.MethodHandles$Lookup.findStatic(MethodHandles.java:777)
        at org.eclipse.golo.cli.command.spi.CliCommand.callRun(CliCommand.java:39)
        at org.eclipse.golo.cli.command.ShebangCommand.execute(ShebangCommand.java:46)
        at org.eclipse.golo.cli.Main.main(Main.java:66)
Caused by: java.lang.VerifyError: Bad local variable type
Exception Details:
  Location:
    test.__$$_sugar_closure_0()Ljava/lang/Object; @42: aload_0
  Reason:
    Type top (current frame, locals[0]) is not assignable to reference type
  Current Frame:
    bci: @42
    flags: { }
    locals: { top, top, 'java/lang/Object', 'java/lang/Object', 'java/lang/Object' }
    stack: { }
  Bytecode:
    0x0000000: 04b8 0043 05b8 0043 ba00 4700 00ba 004e
    0x0000010: 0000 4d2c 03b8 0043 ba00 5100 004e 2c04
    0x0000020: b800 43ba 0051 0000 3a04 2ab0          

        at java.lang.invoke.MethodHandleNatives.resolve(Native Method)
        at java.lang.invoke.MemberName$Factory.resolve(MemberName.java:977)
        at java.lang.invoke.MemberName$Factory.resolveOrFail(MemberName.java:1002)
        ... 5 more
```

this one works as expected: 

```golo
#!/usr/bin/env golosh
module test

function test = {
  let a, b = [1,2]
  return {
    return a
  }
}

function main = |args| {
  let x = test()()
  println(x) # prints 1
}
```

--- a/src/main/java/org/eclipse/golo/compiler/ir/ReferenceTable.java
+++ b/src/main/java/org/eclipse/golo/compiler/ir/ReferenceTable.java
@@ -83,6 +83,11 @@ public final class ReferenceTable implements Scope {
 
   @Override
   public void relink(ReferenceTable parent) {
+    for(LocalReference reference : parent.references()) {
+      if(this.hasReferenceFor(reference.getName())){
+        this.remove(reference.getName());
+      }
+    }
     this.parent = parent;
   }
 

--- a/src/test/resources/for-execution/destruct.golo
+++ b/src/test/resources/for-execution/destruct.golo
@@ -130,6 +130,15 @@ function test_swap = {
   require(c == 1, "err")
 }
 
+# issue #391
+function test_destruct_affectation_inside_closure = {
+  let closure = {
+    let a, b = [1,2]
+    return a
+  }
+  require(closure() == 1, "err")
+}
+
 function main = |args| {
   test_tuple_samesize()
   test_tuple_rest()

